workflow/cron/controller.go

```diff
@@ -123,11 +126,13 @@ func (cc *Controller) processNextCronItem() bool {
 		return true
 	}
 	if !exists {
+		cc.nameEntryIDMapLock.Lock()
 		if entryId, ok := cc.nameEntryIDMap[key.(string)]; ok {
 			log.Infof("Deleting '%s'", key)
 			cc.cron.Remove(entryId)
 			delete(cc.nameEntryIDMap, key.(string))
 		}
+		cc.nameEntryIDMapLock.Unlock()
```

Comment:
I'd like @jessesuen to double check this. Normally we unlock in a defer block - but this is needed if the code may return before the unlock, e.g. due to `return err` or `panic` - I'm not sure either of those apply here.

Reply:
I actually second the unlock in `defer`. We should treat the entire `processNextWorkflowItem` and `processNextCronItem` functions as atomic.

Reply:
Thank you for your comments. I unlock in `defer`.